### DISCUSSION TITLE
How to chose reporting data way?

### DISCUSSION THREAD

**Discussion by Cool-Coding** [apache/skywalking#8907] [Q&A]
As we all know , there are two reporting ways, grpc and kafka. In production env, which do you chose? which one has the better performance?

**Comment by wu-sheng** [comment]
If there is a better way, we have closed the other :) This is all about tradeoffs.

**Reply by Cool-Coding** [reply]
> If there is a better way, we have closed the other :) This is all about tradeoffs.

Recently, We make a performance test with kafka reporting trace data.  the throughput and cost of biz is lower than 25%. maybe the test is not precise because of many containers exist on one physical machine. but we tested two times. the result is the same. so I want to konw which way others commonly use.

**Reply by wu-sheng** [reply]
I think this is not the correct way to test. Sharing resource means you cost 3 times serialization/deserialization(Kafka client + server) and Kafka server's cost on your services.
But, I personally didn't do anything tests similar to this. I am not using Kafka in this scenario.
